src/combinatorics/hales_jewett.lean

```diff
@@ -0,0 +1,298 @@
+/-
+Copyright (c) 2021 David Wärn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Wärn
+-/
+import data.fintype.basic
+import algebra.big_operators.basic
+import data.equiv.fin
+
+/-!
+# The Hales-Jewett theorem
+
+We prove the Hales-Jewett theorem and deduce Van der Waerden's theorem as a corollary.
```

Comment:
Could you recall the statements of the theorems in this file-level docstring (possibly in informal form)?